### ISSUE TITLE
The `attempts` argument for `retry_on` is confusing

### ISSUE THREAD

**Issue by XrXr** [rails/rails#33816]
### Steps to reproduce

Use `retry_on SomeError, attempts: 1` in a job.

### Expected behavior

The documentation for `retry_on` says

https://github.com/rails/rails/blob/5bbaae0d20147d392f1458f1a99d85c245040d04/activejob/lib/active_job/exceptions.rb#L11-L12

So I expect it to retry once. The second line in the doc doesn't support this behavior either; 1 attempt is not "beyond" 1.

### Actual behavior

It doesn't retry when the job raises `SomeError`;  the exception propagates 

### System configuration
**Rails version**: 5.2.1
**Ruby version**: 2.5.1

Here is a minimal test case:

```
#jobs/retry_job.rb
class RetryJob < ApplicationJob
  queue_as :default
  retry_on RuntimeError, attempts: 1

  def perform(*args)
    raise RuntimeError, 'something went wrong'
  end
end
```

```
#jobs/retry_job_test.rb
require 'test_helper'

class RetryJobTest < ActiveJob::TestCase
  test "that it retries" do
    # change `attempts` to 2 for this to pass
    assert_enqueued_jobs 1 do
      RetryJob.perform_now
    end
  end
end
```

**Comment by rafaelfranca** [comment]
I think we can improve the documentation around it but I think the name `attempts` is misleading. I'd change it to `max_retries` and or even change the logic of attempts no count the first attempt.

@matthewd @jeremy what do you think?

**Comment by matthewd** [comment]
I think we have to resolve this in the documentation, because it's not the number of retries, +/- 1 or otherwise: it's not "recover from [some exception] up to N times", it's "recover from [some exception] if we haven't yet failed N times _for any reason_".

**Comment by y-yagi** [comment]
Fixed by #33826
